pandas/tests/indexes/test_base.py

```diff
@@ -125,88 +126,75 @@ def test_construction_list_tuples_nan(self, na_value, vtype):
-        tm.assert_index_equal(result, idx)
-
-        result = pd.Index(idx.astype(object))
-        tm.assert_index_equal(result, idx)
-
-    def test_constructor_from_index_period(self):
-        idx = pd.period_range('2015-01-01', freq='D', periods=3)
-        result = pd.Index(idx)
-        tm.assert_index_equal(result, idx)
-
-        result = pd.Index(idx.astype(object))
-        tm.assert_index_equal(result, idx)
-
-    def test_constructor_from_series_datetimetz(self):
-        idx = pd.date_range('2015-01-01 10:00', freq='D', periods=3,
-                            tz='US/Eastern')
-        result = pd.Index(pd.Series(idx))
-        tm.assert_index_equal(result, idx)
-        assert result.tz == idx.tz
+    @pytest.mark.parametrize("cast_as_obj", [True, False])
+    @pytest.mark.parametrize("idx,has_tz", [
+        (pd.date_range('2015-01-01 10:00', freq='D', periods=3,
+                       tz='US/Eastern'), True),  # datetimetz
+        (pd.timedelta_range('1 days', freq='D', periods=3), False),  # td
+        (pd.period_range('2015-01-01', freq='D', periods=3), False)  # period
+    ])
+    def test_constructor_from_index_dtlike(self, cast_as_obj, idx, has_tz):
+        if cast_as_obj:
+            result = pd.Index(idx.astype(object))
+        else:
+            result = pd.Index(idx)
 
-    def test_constructor_from_series_timedelta(self):
-        idx = pd.timedelta_range('1 days', freq='D', periods=3)
-        result = pd.Index(pd.Series(idx))
         tm.assert_index_equal(result, idx)
-
-    def test_constructor_from_series_period(self):
-        idx = pd.period_range('2015-01-01', freq='D', periods=3)
+        if has_tz:
+            assert result.tz == idx.tz
+
+    @pytest.mark.parametrize("idx,has_tz", [
+        (pd.date_range('2015-01-01 10:00', freq='D', periods=3,
+                       tz='US/Eastern'), True),  # datetimetz
+        (pd.timedelta_range('1 days', freq='D', periods=3), False),  # td
+        (pd.period_range('2015-01-01', freq='D', periods=3), False)  # period
+    ])
+    def test_constructor_from_series_dtlike(self, idx, has_tz):
         result = pd.Index(pd.Series(idx))
         tm.assert_index_equal(result, idx)
 
-    def test_constructor_from_series(self):
+        if has_tz:
+            assert result.tz == idx.tz
 
+    @pytest.mark.parametrize("klass", [Index, DatetimeIndex])
+    def test_constructor_from_series(self, klass):
         expected = DatetimeIndex([Timestamp('20110101'), Timestamp('20120101'),
                                   Timestamp('20130101')])
         s = Series([Timestamp('20110101'), Timestamp('20120101'),
                     Timestamp('20130101')])
-        result = Index(s)
-        tm.assert_index_equal(result, expected)
-        result = DatetimeIndex(s)
+        result = klass(s)
         tm.assert_index_equal(result, expected)
 
+    @pytest.mark.parametrize("klass", [pd.Series, pd.DataFrame])
+    def test_constructor_from_series_freq(self, klass):
```

Comment:
This could arguably be split into two separate tests given the size of the conditional. Open to suggestions

Reply:
yes woudl do that